charmcraft/commands/analyze.py

```diff
@@ -61,23 +61,37 @@ def fill_parser(self, parser):
             help="Produce the result formatted as a JSON string",
         )
 
-    def run(self, parsed_args):
-        """Run the command."""
-        tmpdir = tempfile.mkdtemp()
-        filepath = str(parsed_args.filepath)
+    def _unzip_charm(self, filepath: pathlib.Path) -> pathlib.Path:
+        """Extract the charm content to a temp directory."""
+        tmpdir = pathlib.Path(tempfile.mkdtemp())
         try:
-            zf = zipfile.ZipFile(filepath)
-            zf.extractall(path=tmpdir)
+            zf = zipfile.ZipFile(str(filepath))
+            zf.extractall(path=str(tmpdir))
         except Exception as exc:
             raise CommandError(
-                "Cannot open the indicated charm file {!r}: {!r}.".format(filepath, exc)
+                "Cannot open the indicated charm file {!r}: {!r}.".format(str(filepath), exc)
             )
 
+        # fix permissions as extractall does not keep them (see bpo15795)
```

Comment:
googling bpo15795 found nothing :-/

```suggestion
        # fix permissions as extractall does not keep them (see https://bugs.python.org/issue15795)
```

Very interesting.

Reply:
Ack.